### DISCUSSION TITLE
Resolving different schemes for one service with two pods.

### DISCUSSION THREAD

**Discussion by SpeedyGonzaless** [apollographql/apollo-server#7250] [Q&A]
I have run apollo gateway in k8s and service MyTestService. 

```
const gateway = new ApolloGateway({
    supergraphSdl: new IntrospectAndCompose({
        subgraphs: [
            { name: 'test-serivice', url: 'http://test-service-svc/graphql' }
        ],
        pollIntervalInMs: 5000
    })
});
```

MyTestService has two pods. At first both pods has the same schema:

```
type Query {
    getClient(
        id: Int,
    ): [Client]
}

type Client {
    id: Int,
    name: String
}
```
Then I change Client schema in code to this:
```
type Client {
    id: Int,
    name: String,
    newTestField: String
}
```
And I restart only one pod of MyTestService. So for now, first pod has old schema and second pod has new schema. 
Is it somehow possible not to allow me as a client use field 'newTestField' in request until all the pods will have this new field?

**Comment by trevor-scheer** [comment]
Apollo Studio solves this problem using a schema registry. You push changes to the registry once your service can accept queries with those changes, and that is the schema that your gateway recognizes. We don't recommend `IntrospectAndCompose` for production use and this is one of the main reasons. In any case, the abstract solution here is that you need a single source of truth for your schema that your clients write queries against and is only updated to reflect schema changes when they are valid across all of your pods.
